packages/plugin-ext/src/main/node/resolvers/util.ts

```diff
@@ -0,0 +1,41 @@
+/********************************************************************************
+ * Copyright (C) 2020 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { PluginDeployerResolverContext } from '../../../common/plugin-protocol';
+import * as fs from 'fs';
+import * as path from 'path';
+import { FileUri } from '@theia/core/lib/node';
+import URI from '@theia/core/lib/common/uri';
+
+export async function resolveLocalPluginPath(
```

Comment:
should we use inheritance between the deployers over custom functions ?

If rewritten, we can use fs-extra and await fs.pathExists instead of all `sync` methods

Reply:
Good suggestion, feels much better now: 46b653b9b78c362e4b5f637f8b81e0b7d0190ebc